src/object/cli_shard.c

```diff
@@ -1362,10 +1362,50 @@ struct obj_query_key_cb_args {
 	daos_key_t		*akey;
 	daos_recx_t		*recx;
 	struct dc_object	*obj;
+	struct dc_obj_shard	*shard;
 	struct dtx_epoch	epoch;
 	daos_handle_t		th;
 };
 
+static void
+obj_shard_query_recx_post(struct obj_query_key_cb_args *cb_args,
+			  daos_recx_t *reply_recx)
+{
+	daos_recx_t		*result_recx = cb_args->recx;
+	struct daos_oclass_attr	*oca;
+	uint64_t		 stripe_rec_nr, cell_rec_nr, rx_idx, rx_end;
+	unsigned int		 tgt_idx;
+
+	oca = obj_get_oca(cb_args->obj, false);
+	if (oca == NULL || !DAOS_OC_IS_EC(oca)) {
+		*result_recx = *reply_recx;
+		return;
+	}
+
+	tgt_idx = cb_args->shard->do_pl_shard.po_shard %
+		  obj_ec_tgt_nr(oca);
+	stripe_rec_nr = obj_ec_stripe_rec_nr(oca);
+	cell_rec_nr = obj_ec_cell_rec_nr(oca);
+	if (reply_recx->rx_idx & PARITY_INDICATOR) {
+		rx_idx = (reply_recx->rx_idx & (~PARITY_INDICATOR));
+		D_ASSERT(rx_idx % cell_rec_nr == 0);
+		D_ASSERT(reply_recx->rx_nr % cell_rec_nr == 0);
+		rx_idx = obj_ec_idx_vos2daos(rx_idx, stripe_rec_nr, cell_rec_nr,
+					     0);
+		result_recx->rx_idx = rx_idx;
+		result_recx->rx_nr = stripe_rec_nr *
+				     (reply_recx->rx_nr / cell_rec_nr);
+		return;
+	}
+
+	rx_end = reply_recx->rx_idx + reply_recx->rx_nr;
+	rx_idx = max(reply_recx->rx_idx, rounddown(rx_end - 1, cell_rec_nr));
+	result_recx->rx_nr = rx_end - rx_idx;
+	rx_idx = obj_ec_idx_vos2daos(rx_idx, stripe_rec_nr, cell_rec_nr,
+				     tgt_idx);
+	result_recx->rx_idx = rx_idx;
+}
```

Comment:
probably this should be moved to cli_ec.c. merge to obj_recx_ec2_daos()?

Reply:
this func() only generate one result recx that is different with obj_recx_ec2_daos(), we can consider to merge or reuse some code for them in following patch.